src/cart/crt_rpc.c

```diff
@@ -836,9 +869,9 @@ crt_req_ep_lc_lookup(struct crt_rpc_priv *rpc_priv, crt_phy_addr_t *base_addr)
 				D_GOTO(out, rc);
 			}
 
-			rc = crt_req_get_tgt_uri(rpc_priv, uri);
+			rc = crt_req_fill_tgt_uri(rpc_priv, uri);
 			if (rc != 0) {
-				D_ERROR("crt_req_get_tgt_uri failed, "
+				D_ERROR("crt_req_fill_tgt_uri failed, "
 					"opc: %#x.\n", req->cr_opc);
 				D_GOTO(out, rc);
 			}
```

Comment:
You could potentially move the FREE to here since uri isn't  used outside of the scope of this branch.   You would need another label for that though.

Reply:
yeah, it didnt seem worth it adding another label to me or doing D_FREE inline in error cases in that block